### PR TITLE
feat: auth from storage

### PR DESCRIPTION
**Pull request by olenagerasimova**
part of #1237 

Added `Authentication` implementation to check user password from storage. It will be used with new policy/permissions implementation.

Also, I think it would be better to correct `Authentication` and `Policy` interfaces to return `CompletionStage`. Now implementations use `BlockingStorage` and thus blocking operations occurs in the chain of non-blocking requests. What do you say?

### CODE REVIEW

**Review by ChGen — APPROVED** [review]
@olenagerasimova , looks good to me. As for refactoring those interfaces to non-blocking APIs, I support that, since we intend to support wide range of storage types and authentication methods.

File: src/main/java/com/artipie/auth/AuthFromStorage.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    /**
+     * Ctor.
+     * @param asto Abstract blocking storage
+     */
+    public AuthFromStorage(final BlockingStorage asto) {
+        this.asto = asto;
+    }
+
+    @Override
+    public Optional<User> user(final String name, final String pass) {
```
Comment (dgarus) [inline src/main/java/com/artipie/auth/AuthFromStorage.java:62]: As I understand, `User` class was removed from `http`.
Is this implementation temporary?

Comment (olenagerasimova) [reply src/main/java/com/artipie/auth/AuthFromStorage.java:62]: hm, the class was renamed into `AuthUser` and moved to the upper level, as soon as http is released we will have to correct all the `Authentication` implementations.

Comment (olenagerasimova) [reply src/main/java/com/artipie/auth/AuthFromStorage.java:62]: The corrections will be minimal, just change the class name

**Review by dgarus — APPROVED** [review]
Thanks! lgtm

I suggest making PRs complete because it is hard to review.

**Pull request merged**